core/drivers/scmi-msg/clock.c

```diff
@@ -303,6 +303,12 @@ static void scmi_clock_describe_rates(struct scmi_msg *msg)
 		status =  plat_scmi_clock_rates_array(msg->agent_id, clock_id,
 						      plat_rates, &ret_nb);
 		if (status == SCMI_SUCCESS) {
+			assert((sizeof(p2a) % sizeof(uint32_t)) == 0U);
+			if (((size_t)msg->out % __alignof__(uint32_t)) != 0U) {
```

Comment:
The cast to `(size_t)` looks weird, is it needed? Would `(uintptr_t)` be better?
`!= 0U` is not really needed.

Reply:
Actually i this change seems useless. the rate array, in the buffer referred by `msg->out` is always 32bit alignment by SCMI message layout definition. So this commit "core: scmi-drv: check buffer alignment on clock rate array" could be discarded.